### PR TITLE
Apply the same rules for significant digits in scientific and decimal notation

### PR DESCRIPTION
**Pull request by krlmlr**
@hadley: Can you please take a look at the output changes?

### CODE REVIEW

File: tests/testthat/_snaps/format_integer.md [hunk 1]
```diff
@@ -3,16 +3,21 @@
...
     Code
       pillar(add_special(x), width = 7)
     Output
       <pillar>
         <int>
-       1.00e7
-       1.00e7
-       1.00e7
-      NA     
+        1  e7
```
Comment (hadley) [inline tests/testthat/_snaps/format_integer.md:17]: The space here looks rather odd

File: tests/testthat/_snaps/num.md [hunk 2]
```diff
@@ -48,50 +48,111 @@
...
-      14 1.00e+ 0   1.00e+ 0       1                    1.00 
-      15 1.00e+ 1  10.00e+ 0      10                   10.00 
-      16 1.00e+ 2 100.00e+ 0     100                  100.00 
-      17 1.00e+ 3   1.00e+ 3    1000                    1.00k
-      18 1.00e+ 4  10.00e+ 3   10000                   10.00k
-      19 1.00e+ 5 100.00e+ 3  100000                  100.00k
-      20 1.00e+ 6   1.00e+ 6 1000000                    1.00M
+            sci      eng                   dec       si
+          <sci>    <eng>                 <dec>     <si>
+       1 1.e-13 100.e-15       0.0000000000001    100.f
```
Comment (hadley) [inline tests/testthat/_snaps/num.md:53]: `1.e` looks odd to my eye, and `1 e` is just as weird.

File: tests/testthat/_snaps/format_integer.md (hunk 1 above)
Comment (krlmlr) [reply tests/testthat/_snaps/format_integer.md]: Fixed now.

File: tests/testthat/_snaps/num.md (hunk 2 above)
Comment (krlmlr) [reply tests/testthat/_snaps/num.md]: Fixing in a forthcoming PR.

**Pull request merged**